sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -561,6 +561,11 @@ object ColumnPruning extends Rule[LogicalPlan] {
     case d @ DeserializeToObject(_, _, child) if !child.outputSet.subsetOf(d.references) =>
       d.copy(child = prunedChild(child, d.references))
 
+    case p @ Project(_, s: SerializeFromObject) if p.references != s.outputSet =>
```

Comment:
sorry, one question here: why not `!s.outputSet.subsetOf(p.references)`? I think this is the usual patters used here and I am not sure that equality works well for AttributeSets

Reply:
I think both work. You can see such usage like:

```scala
// prune unrequired references
    case p @ Project(_, g: Generate) if p.references != g.outputSet =>
```

Reply:
yes we are basically assuming here that there cannot be references coming from anywhere else than the `SerializeFromObject` node. Seems reasonable to me as I cannot think of a case where this is not true. Thanks!